src/main.rs

```diff
@@ -240,6 +271,129 @@ async fn main() -> Result<()> {
                 .collect::<Vec<_>>();
             println!("Groups found: {:?}", group_ids)
         }
+        CommandLineArgs::FileEncrypt {
+            filename: infile,
+            device_path,
+            users: user_ids,
+            groups: group_ids,
+            output: maybe_output,
+        } => {
+            let file = std::fs::read(infile.clone())?;
+            println!("Read in file {:?}", infile);
+            let sdk = initialize_sdk_from_file(&device_path).await?;
+            let output = match maybe_output {
```

Comment:
It seems like this match could be broken out to be a function which could be unit tested.

Reply:
Done and tested